src/etc/inc/interfaces.inc

```diff
@@ -3006,44 +3021,76 @@ function interface_dhcpv6_prepare($interface = 'wan', $wancfg)
+                "/var/run/dhcp6c.pid",                
+            )
+        );
+    }
+
+    // Add the interfaces on the end, as there may be more than one interface - multiple WAN dhcp6c.
+    
+    foreach (legacy_config_get_interfaces(['enable' => true, 'virtual' => false]) as $if => $wancfg) {
+        // check the interface is up
+        $realif = get_real_interface($if,'inet6');
+        $intf_stats = legacy_interface_stats($realif);
+        if($intf_stats['link state'] != 2) {
+            continue;
+        }
+        if (!empty($wancfg['ipaddrv6']) && $wancfg['ipaddrv6'] == 'dhcp6' && $wancfg['enable'] == true) {
+            
+            $dhcp6ccommand = exec_safe("{$dhcp6ccommand} %s", get_real_interface($if, 'inet6'));
+        }
+    }
 
-    $dhcp6ccommand = exec_safe(
-        "/usr/local/sbin/dhcp6c %s -c %s -p %s %s",
-        array(
-            '-' . (empty($wancfg['adv_dhcp6_debug']) ? 'd' : 'D' ) . (!empty($wancfg['dhcp6norelease']) ? 'n' : ''),
-            "/var/etc/dhcp6c_{$interface}.conf",
-            "/var/run/dhcp6c_{$wanif}.pid",
-            "{$wanif}"
-        )
-    );
+    log_error("dhcp6c command = {$dhcp6ccommand}"); // debug purposes only
 
     $rtsoldscript = "#!/bin/sh\n";
     $rtsoldscript .= "# this file was auto-generated, do not edit\n";
     $rtsoldscript .= "if [ -n \"\${2}\" ]; then\n";
     $rtsoldscript .= "\techo \${2} > /tmp/{$wanif}_routerv6\n";
     $rtsoldscript .= "\techo \${2} > /tmp/{$wanif}_defaultgwv6\n";
     $rtsoldscript .= "fi\n";
-    $rtsoldscript .= "if [ -f /var/run/dhcp6c_{$wanif}.pid ]; then\n";
-    $rtsoldscript .= "\t/usr/bin/logger -t dhcp6c \"RTSOLD script - Sending SIGHUP to dhcp6c for interface {$interface}({$wanif})\"\n";
-    $rtsoldscript .= "\t/bin/pkill -HUP -F /var/run/dhcp6c_{$wanif}.pid\n";
+    $rtsoldscript .= "val=`pgrep dhcp6c`\n";
+    $rtsoldscript .= "pid=`cat /var/run/dhcp6c.pid`\n";
+    $rtsoldscript .= "if [ \$pid -eq \$val and \$pid -gt 0 ]; then\n";
+    $rtsoldscript .= "\t/usr/bin/logger -t dhcp6c \"RTSOLD script - Sending SIGHUP to dhcp6c client\"\n";
+    $rtsoldscript .= "\t/bin/pkill -HUP -F \$pid\n";
     $rtsoldscript .= "else\n";
-    $rtsoldscript .= "\t/usr/bin/logger -t dhcp6c \"RTSOLD script - Starting dhcp6 client for interface {$interface}({$wanif})\"\n";
+    $rtsoldscript .= "\t/usr/bin/killall -9 -c dhcp6c\n";
+    $rtsoldscript .= "if [ -f '/var/run/dhcp6c.pid' ]; then\n";
+    $rtsoldscript .= "\trm '/var/run/dhcp6c.pid'\n";
+    $rtsoldscript .= "fi\n";    
+    $rtsoldscript .= "\t/usr/bin/logger -t dhcp6c \"RTSOLD script - Starting dhcp6 client\"\n";
     $rtsoldscript .= "\t$dhcp6ccommand\n";
     $rtsoldscript .= "fi\n";
 
     @file_put_contents("/var/etc/rtsold_{$wanif}_script.sh", $rtsoldscript);
     @chmod("/var/etc/rtsold_{$wanif}_script.sh", 0755);
 }
 
-function DHCP6_Config_File_Basic($interface, $wancfg, $wanif, $id = 0)
+function DHCP6_Config_File_Basic($realif, $interface, $wancfg, $id = 0)
```

Comment:
basically here we reverse variable order and rename them for no benefit of the function's code.